bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/FindReplaceOverlay.java

```diff
@@ -226,11 +228,22 @@ public void controlResized(ControlEvent e) {
 		}
 	};
 
-	private FocusListener overlayFocusListener = FocusListener.focusLostAdapter(e -> {
-			findReplaceLogic.activate(SearchOptions.GLOBAL);
-			searchInSelectionButton.setSelection(false);
-	});
+	/**
+	 * Updates the find replace dialog on activation changes.
+	 */
+	class ActivationListener extends ShellAdapter {
+		@Override
+		public void shellActivated(ShellEvent e) {
+			// Do nothing
+		}
 
+		@Override
+		public void shellDeactivated(ShellEvent e) {
+			removeSearchScope();
+		}
+	}
+	
+	private ActivationListener overlayFocusListener = new ActivationListener();
```

Comment:
You could also assign an anonymous class with the according implementation instead of defining a new `ActivationListener` class, i.e., `private ShellListener overlayDeactivationListener = new ShellAdapter() { ...};
```suggestion
	private ActivationListener overlayDeactivationListener = new ActivationListener();
```

Reply:
Good points, I agree that they improve the code quality!

Addressed in https://github.com/eclipse-platform/eclipse.platform.ui/commit/f4a575c53678d57f08428e30573dc666f79cebbd